### ISSUE TITLE
Feature Request: Choose the mode used for generating model summary 

### ISSUE THREAD

**Issue by princyok** [TylerYep/torchinfo#121]
**Current state**
Version: 1.6.3

Currently, `torchinfo.summary` generates the summary of a model using strictly the evaluation mode. [This is the line](https://github.com/TylerYep/torchinfo/blob/fb4605de45eed50535f1a144e6d63bd8416774d5/torchinfo/torchinfo.py#L271) where that decision is forced.

**Request**
Allow the user to choose which mode to use for generating the summary. 

The fix is simple. Something like this:
````python
def summary(..., mode="eval", ...):
    ...
    ...
    summary_list = forward_pass(..., mode, ...)
    ...
    ...
def forward_pass(..., mode, ...):
    ...
    ...
    if mode == "eval":
       model.eval()
    elif mode == "train":
        model.train() 
    ...
    ...

**Comment by TylerYep** [comment]
Can you explain the use case for this? In general, it is not advised to actually train your model using torchinfo, and we want to avoid any side effects caused by calling summary().

**Comment by princyok** [comment]
@TylerYep When the forward pass operations are drastically different for the different modes, but I still want to get the summaries for both modes of the model.

**Comment by TylerYep** [comment]
Got it, thanks! I'm working on a fix now.

**Comment by TylerYep** [comment]
Implemented and released in v1.6.4. Thanks!
